equinix/resource_fabric_network.go

```diff
@@ -0,0 +1,248 @@
+	location := locationToFabric(schemaLocation)
+	project := v4.Project{}
+	schemaProject := d.Get("project").(*schema.Set).List()
+	if len(schemaProject) != 0 {
+		project = projectToFabric(schemaProject)
+	}
+	netType := v4.NetworkType(d.Get("type").(string))
+	netScope := v4.NetworkScope(d.Get("scope").(string))
+
+	createRequest := v4.NetworkPostRequest{
+		Name:          d.Get("name").(string),
+		Type_:         &netType,
+		Scope:         &netScope,
+		Location:      &location,
+		Notifications: notifications,
+		Project:       &project,
+	}
+
+	fabricNetwork, _, err := client.NetworksApi.CreateNetwork(ctx, createRequest)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	d.SetId(fabricNetwork.Uuid)
+
+	if _, err = waitUntilNetworkIsProvisioned(d.Id(), meta, ctx); err != nil {
+		return diag.Errorf("error waiting for Network (%s) to be created: %s", d.Id(), err)
+	}
+
+	return resourceNetworkRead(ctx, d, meta)
+}
+
+func resourceNetworkRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	fabricNetwork, _, err := client.NetworksApi.GetNetworkByUuid(ctx, d.Id())
+	if err != nil {
+		log.Printf("[WARN] Fabric Network %s not found , error %s", d.Id(), err)
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.FromErr(err)
+	}
+	d.SetId(fabricNetwork.Uuid)
+	return setNetworkMap(d, fabricNetwork)
+}
+
+func setNetworkMap(d *schema.ResourceData, nt v4.Network) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	err := equinix_schema.SetMap(d, map[string]interface{}{
+		"name":          nt.Name,
+		"type":          nt.Type_,
+		"scope":         nt.Scope,
+		"state":         nt.State,
+		"operation":     networkOperationToTerra(nt.Operation),
+		"change":        simplifiedNetworkChangeToTerra(nt.Change),
+		"location":      locationToTerra(nt.Location),
+		"notifications": notificationToTerra(nt.Notifications),
+		"project":       projectToTerra(nt.Project),
+		"change_log":    changeLogToTerra(nt.ChangeLog),
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return diags
+}
+
+func resourceNetworkUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.Config).FabricClient
+	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	dbConn, err := waitUntilNetworkIsProvisioned(d.Id(), meta, ctx)
+	if err != nil {
+		if !strings.Contains(err.Error(), "500") {
+			d.SetId("")
+		}
+		return diag.Errorf("either timed out or errored out while fetching Fabric Network for uuid %s and error %v", d.Id(), err)
+	}
+	// TO-DO
+	update, err := getNetworkUpdateRequest(dbConn, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updates := []v4.NetworkChangeOperation{update}
+	_, res, err := client.NetworksApi.UpdateNetworkByUuid(ctx, updates, d.Id())
```

Comment:
`res` from Fabric_Go SDK should be left out of errors. It will only show the headers of the request and doesn't provide value to the end user. Just clutters the error output. Please follow the pattern set for error output on Fabric Resources defined here: https://github.com/equinix/terraform-provider-equinix/pull/449